pkg/services/store/object/http.go

```diff
@@ -47,6 +51,9 @@ func (s *httpObjectStore) RegisterHTTPRoutes(route routing.RouteRegister) {
 	route.Get("/history/*", reqGrafanaAdmin, routing.Wrap(s.doGetHistory))
 	route.Get("/list/*", reqGrafanaAdmin, routing.Wrap(s.doListFolder)) // Simplified version of search -- path is prefix
 	route.Get("/search", reqGrafanaAdmin, routing.Wrap(s.doSearch))
+
+	// File upload
+	route.Post("/upload", reqGrafanaAdmin, routing.Wrap(s.doUpload))
```

Comment:
I'm quite concerned about starting to use the object store for any kind of file right now. can we defer this?

Uploading the files to blog storage and keeping a reference of them in the object store would probably be my suggestion.

Reply:
🤣 lots of catch 22s here!!!!  What about the existing `file` sql store?!  that is pretty much identical to this table (without the summary info that gets saved in the same transaction)

I agree blob storage is the best path -- but I think that comes later, after we agree on an object store interface that is comfortable with "objects"

For sure we need configurable rate/quota limiting.  But hard to say uploading images+or csv is worse than 20mb dashboards 🤷

Reply:
We have 20mb dashboard json blobs ? How is that possible? :)

Reply:
panels with embedded data (testdata etc) / snapshots

When rolling out search in cloud -- the spikes we would see had more to do with large dashboards than with number of dashboards.